### PR TITLE
Happy thought app (Tara)

### PR DESCRIPTION
**Pull request by trangtara**
Link to my deployed project:

My reflections on how this week's project turned out:

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: code/src/App.js [hunk 1]
```diff
@@ -1,9 +1,116 @@
-import React from 'react'
+import React,{ useState, useEffect } from 'react';
+
+import Header from './Component/Header';
+import ThoughtSubmitForm from 'Component/ThoughtSubmitForm';
+import MessageList from 'Component/MessageList';
+import { API_URL } from './constants';
```
Comment (karolinalbinsson) [inline code/src/App.js:6]: Nice!

File: code/src/App.js [hunk 2]
```diff
@@ -1,9 +1,116 @@
...
+import './index.css';
+
+//there are 3 erros found in the response relating to empty, too short or too long messages.
+const THOUGHT_ERRORS = [
```
Comment (karolinalbinsson) [inline code/src/App.js:10]: This was a really nice touch!

File: code/src/App.js [hunk 3]
```diff
@@ -1,9 +1,116 @@
...
+          });
+          if (messageItem) {
+            throw new Error(messageItem.message);
+          } else {
+            //this else if is just to make sure if there are any errors that are not the same with the three identified errors
+            throw new Error("Something goes wrong. Call someone to share your thought then")
+          }
+        }
+
+        // When post request was successful
```
Comment (karolinalbinsson) [inline code/src/App.js:72]: Good explanatory comments 👍

File: code/src/Component/Header.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import  '../Style/Header.css';
+
+const Header = () => {
+  return (
+  <div className="header">
+    <h1 className="header-title">
+      <div className="header-title-row-1">
+        <span className="header-title-1">Happy</span> <span className="header-title-2">Thought</span>
```
Comment (karolinalbinsson) [inline code/src/Component/Header.js:9]: By just looking at the code here it was difficult to understand why you've used span inside h1. When looking at your site it is quite clear - but consider using some self-explaining class names.

File: code/src/Component/LikeButton.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    onLikeClick(thought._id)
+  };
+
+  return (
+    <div className="like-button-container">
+      <button 
+        className={classNames.join(' ')}
+        type="button"
+        onClick={() => clickLikeButton()}>
+        <span className="heart-icon-like" aria-label="heart emoji" role="img">&#10084;&#65039;</span>
```
Comment (karolinalbinsson) [inline code/src/Component/LikeButton.js:30]: Good usage of span with aria-label on emojis :)

File: code/src/App.js [hunk 6]
```diff
@@ -1,9 +1,116 @@
...
+        "Content-Type": "application/json"
+      },
+      method: "POST",
+      });
+      //Here we will not perform any response fetch because we just need to post the "Like"
+  };
+
   return (
     <div>
-      Find me in src/app.js!
```
Comment (karolinalbinsson) [inline code/src/App.js:6]: You could remove this line

File: code/src/Component/LikeButton.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import React, { useEffect, useState } from 'react';
+
+import '../Style/LikeButton.css';
+import '../Style/HeartIcon.css';
+
+// This component is to update the like button on the messages. For better performance and prevent fetching the whole messageList on each Like, the likeCount will be updated in the local state rather than to the API.
+const LikeButton = ({ thought, onLikeClick }) => {
+  const [classNames, setClassNames] = useState([]);
+  const [likeCount, setLikeCount] = useState(thought.hearts);
```
Comment (karolinalbinsson) [inline code/src/Component/LikeButton.js:9]: Nice usage of initial state!

File: code/src/Style/Header.css [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  right: 0;
+  top: 0;
+  width: 100vw;
+}
+
+.header-title {
+  align-items: center;
+  display: flex;
+  flex-direction: column;
+  font-family: 'Capriola', sans-serif;
```
Comment (karolinalbinsson) [inline code/src/Style/Header.css:14]: Great that you are using a fallback font 👍

File: code/src/Style/Container.css [hunk 9]
```diff
@@ -0,0 +1,32 @@
+.container {
```
Comment (karolinalbinsson) [inline code/src/Style/Container.css:1]: I like that you have divided your CSS in different files, it makes it easy for me to follow.

File: code/src/Style/MainForm.css [hunk 10]
```diff
@@ -0,0 +1,7 @@
+
+.main-form {
```
Comment (karolinalbinsson) [inline code/src/Style/MainForm.css:2]: Could margin:auto be replaced by using display:flex on parent element?

File: code/src/Style/ThoughtSubmitForm.css [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+  border-radius: 20px;
+  border: none;
+  color: #ffffff;
+  display: flex;
+  align-items: center;
+  font-family:  'Capriola', sans-serif;
+  font-size: 13px;
+  margin-top: 10px;
+  padding: 5px 15px;
+
```
Comment (karolinalbinsson) [inline code/src/Style/ThoughtSubmitForm.css:12]: blank line, remove

File: code/src/Style/ThoughtSubmitForm.css [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+.word-count {
+  color: blue;
+  flex: 1;
+  font-family: 'Raleway', sans-serif;
+  font-weight: 500;
+  font-size: 13px;
+  text-align: end;
+}
```
Comment (karolinalbinsson) [inline code/src/Style/ThoughtSubmitForm.css:42]: I think you could be more consistent using blank spaces inbetween your css-classes - either no spaces or one space.

File: code/src/Style/ThoughtSubmitForm.css [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+.count-turn-red {
+  font-weight: bold;
+  color: darkmagenta;
+}
+
+.cross-sign {
+  color: red;
```
Comment (karolinalbinsson) [inline code/src/Style/ThoughtSubmitForm.css:49]: consider also being consistant in using plain-text for colors or HEX. You are using HEX in a lot of other Css-classes.

File: code/public/index.html [hunk 14]
```diff
@@ -4,6 +4,13 @@
   <head>
     <meta charset="utf-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta property="og:title" content="Happiness is meant to be shared">
```
Comment (karolinalbinsson) [inline code/public/index.html:7]: Nice usage of og-tags

File: code/src/App.js [hunk 15]
```diff
@@ -1,9 +1,116 @@
...
+//there are 3 erros found in the response relating to empty, too short or too long messages.
+const THOUGHT_ERRORS = [
+  { kind: 'required', message: 'Please type in your happy thought' },
+  { kind: 'minlength', message: 'Message should be longer than 5 characters' },
+  { kind: 'maxlength', message: 'Oops! Can you keep the message shorter than 140 characters, please.' }
+];
 
 export const App = () => {
+  const [thought, setThought] = useState(""); //the thought state needs to be here in the App.js instead in the child component to handle the erro update while hitting submit button.
+  const [thoughts, setThoughts] = useState([]);
```
Comment (karolinalbinsson) [inline code/src/App.js:18]: these state variables are quite similar in the naming.. Maybe it could be more clear.

File: code/src/App.js [hunk 16]
```diff
@@ -1,9 +1,116 @@
...
+        return res.json();
+      })
+      .then((data) => {
+        setThoughts(data)
+      });
+  };
+
+  const handleThoughtChange = (newThought) => {
+    setThought(newThought);
+  };
```
Comment (karolinalbinsson) [inline code/src/App.js:38]: You are being consistent in using semi-colons 👍

File: code/src/App.js [hunk 17]
```diff
@@ -1,9 +1,116 @@
...
+
+  // errorMassage will be cleared and Textarea for thought input should be cleared too.
+  const clearForm = () => {
+    setErrorMessage(null);
+    setThought("");
+  };
+
+  const handleThoughtSubmit = (thought) => {
+
```
Comment (karolinalbinsson) [inline code/src/App.js:47]: why the empty line here?

File: code/src/Component/LikeButton.js [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+
+  //this function is to change the add the "pink-button" class when the likeCount is > 0
```
Comment (karolinalbinsson) [inline code/src/Component/LikeButton.js:11]: Niiiice!!

File: code/src/Style/ThoughtSubmitForm.css [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+  color: blue;
+  flex: 1;
+  font-family: 'Raleway', sans-serif;
+  font-weight: 500;
+  font-size: 13px;
+  text-align: end;
+}
+.count-turn-red {
+  font-weight: bold;
+  color: darkmagenta;
```
Comment (karolinalbinsson) [inline code/src/Style/ThoughtSubmitForm.css:45]: Could you use HEX here instead?

**Review by karolinalbinsson — CHANGES_REQUESTED** [review]
Hi Tara!

Good job on this week's assignment! 
I think your code is clean and well-indented. It is very easy to read. You are being consistent in usage of semi-colons and you have good indentation. You are also good at writing comments that explain the functions, keep up the good work!

Over all I think you have met the requirements for the project. Although you have some unneccesary commented-out code that you could remove. 

Also consider looking closely to the project brief regarding design - your design is really cute but it is not like the design in the project brief that we are instructed to follow :) 

I enjoyed reading your code because it's well structured and I am taking that inspiration with me into coming projects :)
Before approving your code i'd like to see that you remove commented-out code. 

Good job! ❤️ 
/Karolin

File: code/src/Style/ThoughtSubmitForm.css (hunk 11 above)
Comment (trangtara) [reply code/src/Style/ThoughtSubmitForm.css:12]: removed!

File: code/src/Style/ThoughtSubmitForm.css (hunk 13 above)
Comment (trangtara) [reply code/src/Style/ThoughtSubmitForm.css:49]: colors changed from plain text to HEX!

File: code/src/App.js (hunk 17 above)
Comment (trangtara) [reply code/src/App.js:47]: Hehe, i dont know why. Removed space! Thanks!

File: code/src/Style/ThoughtSubmitForm.css (hunk 19 above)
Comment (trangtara) [reply code/src/Style/ThoughtSubmitForm.css:45]: solved!

File: code/public/index.html [hunk 20]
```diff
@@ -4,6 +4,13 @@
...
+    <meta property="og:description" content="What makes you happy today?">
+    <meta property="og:image" width="640" height="426" content="https://thinkhappylivehappy.netlify.app/images/og-image.jpg">
+    <meta property="og:url" content="https://thinkhappylivehappy.netlify.app/">
+    <link href="https://fonts.googleapis.com/css2?family=Raleway:wght@300&family=Source+Sans+Pro:wght@300&family=Capriola&display=swap" rel="stylesheet">
+    <link rel="icon" type="image/png" href="./images/icons8-love-letter-48.png">
+
     <!--
```
[the remark this answers is not in the document]
Comment (trangtara) [reply code/public/index.html:14]: comment removed!

**Comment by trangtara** [comment]
> Hi Tara!
> 
> Good job on this week's assignment!
> I think your code is clean and well-indented. It is very easy to read. You are being consistent in usage of semi-colons and you have good indentation. You are also good at writing comments that explain the functions, keep up the good work!
> 
> Over all I think you have met the requirements for the project. Although you have some unneccesary commented-out code that you could remove.
> 
> Also consider looking closely to the project brief regarding design - your design is really cute but it is not like the design in the project brief that we are instructed to follow :)
> 
> I enjoyed reading your code because it's well structured and I am taking that inspiration with me into coming projects :)
> Before approving your code i'd like to see that you remove commented-out code.
> 
> Good job! ❤️
> /Karolin

**Review by karolinalbinsson — APPROVED** [review]

**Pull request closed without merging**